iambic/core/utils.py

```diff
@@ -144,14 +147,17 @@ async def gather_templates(repo_dir: str, template_type: str = None) -> list[str
         if template_type
         else NOQ_TEMPLATE_REGEX
     )
+    # since multiple glob pattern can potential intersect, we use a set data structure
+    # to suppress any duplicate for defensive measure
     # Support both yaml and yml extensions for templates
-    file_paths = glob.glob(f"{repo_dir}/**/*.yaml", recursive=True)
-    file_paths += glob.glob(f"{repo_dir}*.yaml", recursive=True)
-    file_paths += glob.glob(f"{repo_dir}/**/*.yml", recursive=True)
-    file_paths += glob.glob(f"{repo_dir}*.yml", recursive=True)
+    file_path_set = set()
+    # >>> glob.glob('**/*.txt', recursive=True)
+    # ['2.txt', 'sub/3.txt']
+    file_path_set.update(glob.glob(f"{repo_dir}/**/*.yaml", recursive=True))
+    file_path_set.update(glob.glob(f"{repo_dir}/**/*.yml", recursive=True))
 
     file_paths = await gather_limit(
-        *[file_regex_search(fp, regex_pattern) for fp in file_paths],
+        *[file_regex_search(fp, regex_pattern) for fp in file_path_set],
```

Comment:
See comment above using pathlib.match instead

Reply:
At this point, file_regex_search is actually looking inside the file contents, so I don't think we will use pathlib.match